src/main/java/io/confluent/idesidecar/restapi/kafkarest/RecordsV3ApiImpl.java

```diff
@@ -63,18 +65,33 @@ public Uni<ProduceResponse> produceRecord(
       @HeaderParam(CONNECTION_ID_HEADER) String connectionId,
       @PathParam("cluster_id") String clusterId,
       @PathParam("topic_name") String topicName,
+      @QueryParam("dry_run") @DefaultValue("false") boolean dryRun,
       @Valid ProduceRequest produceRequest
   ) {
-    return uniItem(ProduceContext.fromRequest(connectionId, clusterId, topicName, produceRequest))
+    var uptoDryRun = uniItem(ProduceContext.fromRequest(connectionId, clusterId, topicName, produceRequest))
         .chain(this::ensureTopicPartitionExists)
         .chain(this::ensureKeyOrValueDataExists)
         .chain(this::fetchClients)
         .chain(this::getSchemas)
         .chain(this::serialize)
-        .chain(this::sendSerializedRecord)
         .onFailure(RuntimeException.class)
-        .transform(this::unwrapRootCause)
-        .map(this::toProduceResponse);
+        .transform(this::unwrapRootCause);
+
+    if (dryRun) {
+      return uptoDryRun
+          // Return a 200 response with the cluster and topic name set.
+          // This should indicate that the request as a whole is valid.
+          .map(c -> ProduceResponse
+              .builder()
+              .clusterId(c.clusterId)
+              .topicName(c.topicName)
+              .build()
+          );
+    } else {
+      return uptoDryRun
+          .chain(this::sendSerializedRecord)
```

Comment:
Do we need to chain this with `.onFailure(RuntimeException.class).transform(this::unwrapRootCause)`?

Reply:
Ah, yes. Thanks